### PR TITLE
Stats: Fix Stats view stuck on Insights tab if the grow audience card is showing

### PR DESCRIPTION
**Pull request by staskus**
## Description

Fixes an issue, when the stats view is stuck on the Insights tab and switching tabs, doesn't work.

## Root cause

The issue is caused by [Stay on insights tab if Grow Audience card is showing](https://github.com/wordpress-mobile/WordPress-iOS/pull/17332) change. 
- When `isGrowAudienceShowing` is `true` we don't save the selected tab to user defaults. 
- However, it fails in cases when we already have the selected tab saved. `updatePeriodView` incorrectly assumes that the wrong tab is selected and doesn't set `periodTableViewController` in `pageViewController`.

## Solution

When `isGrowAudienceShowing` is `true`, explicitly set selected tab to `insights`.

### Case 1:
1. Fresh install app
2. Open Stats
3. Notice "A tip to grow your audience" is shown
4. Dismiss this tip
5. Select any of Days/Weeks/Months/Years
6. Come back to Menu
7. Select Stats again
8. Previously selected Days/Weeks/Months/Years tab should open
9. Open Insights
10. Notice another "A tip to grow your audience" is shown
11. Select any of Days/Weeks/Months/Years
12. The tabs should be switching successfully

## Regression Notes

1. Potential unintended areas of impact

[Stay on insights tab if Grow Audience card is showing](https://github.com/wordpress-mobile/WordPress-iOS/pull/17332)

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing scenarios

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### Before

https://user-images.githubusercontent.com/4062343/194577145-c89208bf-1366-4883-820c-db8d580d40ff.MP4

### After

https://user-images.githubusercontent.com/4062343/194577250-1e8a7d01-d94d-48f0-b6e4-62171c3b3557.MP4

### CODE REVIEW

**Comment by staskus** [comment]
Also cherry-picked bundler fix https://github.com/wordpress-mobile/WordPress-iOS/pull/19409

**Review by guarani — APPROVED** [review]
Thanks for finding out how to reproduce this bug and for fixing it! 🙇 

These are the steps I used to reproduce (taken from p5T066-3CX-p2#comment-13674) and my understanding of the issue:
1. Open Stats
2. Dismiss the "A tip to grow your audience" tip card that's shown
3. Switch to the Months tab (the tab selection is persisted)
4. Close Stats
5. Open Stats and the Months tab is shown (correct)
6. Switch to the Insights tab and a different tip is shown
  - The new tab selection wasn't being persisted here, but with this PR it's correctly set to Insights
8. Try to switch to the Month tab and notice the contents of the Insights tab are still show
  - The "old" and "new" tab selections are both "Months", so the logic [here](https://github.com/wordpress-mobile/WordPress-iOS/blob/1f63600160d7c2fb125751f6c80f0a23d7a9794c/WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift#L252-L256) doesn't set the view controller to show the screen that corresponds to Months

I agree that the approach you've taken which persists the tab selection as Insights if there is a tip card to be shown – or the specific Day/Week/Year tab if there is no tip card – is a good approach. The logic to take the user to the Insights tab if there is a tip card available (https://github.com/wordpress-mobile/WordPress-iOS/pull/17332) – no matter what tab they were previously on – is still working.

File: Gemfile.lock [hunk 1]
```diff
@@ -396,4 +396,4 @@ DEPENDENCIES
   xcpretty-travis-formatter
 
 BUNDLED WITH
-   2.3.22
+   2.3.23
```
Comment (guarani) [inline Gemfile.lock:399]: nitpick: Does CI pass without this change?

Comment (staskus) [reply Gemfile.lock:399]: Unfortunately, no. https://github.com/wordpress-mobile/WordPress-iOS/pull/19409

**Comment by staskus** [comment]
@guarani will the release branch be merged to the trunk once the release is done? Or should I make an additional PR to merge this change to trunk?

**Comment by guarani** [comment]
> @guarani will the release branch be merged to the trunk once the release is done? 

Yep, you don't need to do that.

**Comment by twstokes** [comment]
Thank you @staskus and @guarani for taking care of this! I was AFK, otherwise I would've been happy to review. 🙇

**Comment by mokagio** [comment]
@staskus  this has been bundled as part of [20.9 beta 1 (20.9.0.1)](https://github.com/wordpress-mobile/WordPress-iOS/pull/19438).

Thanks for your work 🙌

**Pull request merged**